netbox/dcim/models/devices.py

```diff
@@ -1277,6 +1277,11 @@ def save(self, *args, **kwargs):
                 if not disable_replication:
                     create_instances.append(template_instance)
 
+            for component in create_instances:
+                # Set default values for any applicable custom fields
+                if cf_defaults := CustomField.objects.get_defaults_for_model(component_model):
```

Comment:
Can we flip this with the statement above to avoid calling it for each individual component?

Reply:
Ooh, good call. My eye skimmed over `component_model` and read it as `component`.